examples/src/ThemeExample.js

```diff
@@ -0,0 +1,40 @@
+// @flow
+import React from 'react';
+import styled, { useTheme, ThemeProvider } from 'styled-components';
+import merge from 'lodash/merge';
+
+import BoxButton from '../../src/components/button';
+import { generateContrastColors } from '../../src/components/theme';
+
+const Button = styled(BoxButton)`
```

Comment:
Should Button.js export this?

Reply:
Basically, if theme provider exists, then spit out the styled button, otherwise spit out the regular button.

Reply:
I'm looking to make a design doc around this to see how we might be able to provide a default theme into each of the BUIE components and allow it to be overridden by the provider context. This way, we'll have a single, consistent way of theming our components.

Doing this would mean the override path for styling would be `.btn` as the theming would be a hash of the styled component styling.

That implementation would mean, the export from the Button component would contain the styled component instead of a separate export.

e.g.
```
const StyledButton = button.styled`
  background: ${props => props.theme.primary.background}
`

...

return (
    <StyledButton
        className={styleClassName}
    >
        <span className="btn-content">{children}</span>
    </StyledButton>
)
```

Reply:
Btw one should still discuss alternatives like the older approach now that one knows (before we were basing this approach on the premise that more components will need to be styled and after talking with deisn sounds like thats not the case) that nothing but the button and left bar will be themed. And since only those two, does it still make sense to introduce styled components or is that an overkill compared to prior style override approach.